Настроить окружение для юнит-тестов (#13)

birthdaysgift opened the pull request:
Так как мы хотим проверять тестами весь код, попадающий в репозиторий, то, перед тем как мы начнём писать какой-либо код, необходимо настроить локальное окружение для написания и запуска тестов.

В рамках этой задачи необходимо:

- добавить `pytest` и `pytest-cov` в `pyproject.toml`
- сгенерировать новый `poetry.lock` файл и установить зависимости локально
- добавить простую функцию и тест для неё, чтобы проверить, что `pytest` работает
- настроить branch-coverage и минимальное покрытие 100%
- при необходимости добавить в конфиги линтеров исключения для тестового кода
- добавить инструкции по запуску тестов и получению coverage-репортов в `README.md`

***

PS. Все остальные библиотеки, описанные в общем плане MVP (типа `factory-boy`, `pytest-mock` и пр.) будем добавлять по мере необходимости в соответствующих задачах.

Rey-Lane commented on .coveragerc:
```diff
@@ -0,0 +1,18 @@
+    if TYPE_CHECKING:
+    pragma: no cover
+
+fail_under = 100
```
Как-то смущает меня этот параметр
Хоть у него и благие намерения))

birthdaysgift replied:
Да не боись, всё будет огонь :fire: 

К тому же если мы что-то не хотим покрывать, то у нас всегда есть `# pragma: no cover`

birthdaysgift committed:
13: Настроить окружение для юнит-тестов

Так как мы хотим проверять тестами весь код, попадающий в репозиторий, то, перед тем как мы начнём писать какой-либо код, необходимо настроить локальное окружение для написания и запуска тестов.

В рамках этой задачи необходимо:

- добавить `pytest` и `pytest-cov` в `pyproject.toml`
- сгенерировать новый `poetry.lock` файл и установить зависимости локально
- добавить простую функцию и тест для неё, чтобы проверить, что `pytest` работает
- настроить branch-coverage и минимальное покрытие 100%
- при необходимости добавить в конфиги линтеров исключения для тестового кода
- добавить инструкции по запуску тестов и получению coverage-репортов в `README.md`

***

PS. Все остальные библиотеки, описанные в общем плане MVP (типа `factory-boy`, `pytest-mock` и пр.) будем добавлять по мере необходимости в соответствующих задачах.

birthdaysgift commented:
Коммит, на который ссылается пулл реквест не существует, так как мы меняли соглашение по именованию коммитов и ребейзили `develop` ветку. Подробнее об этом можно почитать в [обсуджении](https://github.com/week-password/wisher-backend/discussions/3#discussioncomment-5640598).

Хэш нового коммита, содержащего точную копию изменений внесённых этим пулл-реквестом: 2fd4126
